### DISCUSSION TITLE
Version 0.6.5

### DISCUSSION THREAD

**Discussion by IvyZX** [google/flax#2857] [General]
## What's Changed
* Internal change by @copybara-service in https://github.com/google/flax/pull/2826
* Cache .mypy_cache on CI by @cgarciae in https://github.com/google/flax/pull/2819
* improve nn.map_variables docstrings by @cgarciae in https://github.com/google/flax/pull/2818
* Render non-array variables in tabulate by @cgarciae in https://github.com/google/flax/pull/2769
* Improve nn.map_variables example by @cgarciae in https://github.com/google/flax/pull/2831
* Introduce relaxed naming policy behind a feature flag. by @levskaya in https://github.com/google/flax/pull/2833
* Bumps minimal JAX version from 0.3.16 to 0.3.24 and cleans up some code. by @copybara-service in https://github.com/google/flax/pull/2840
* import initializers module directly by @chiamp in https://github.com/google/flax/pull/2830
* Re-box axis metadata in DenseGeneral's init functions by @IvyZX in https://github.com/google/flax/pull/2843
* Update jax dependency by @jheek in https://github.com/google/flax/pull/2827
* replaced `zeros` and `ones` initializers with `zeros_init` and `ones_init` builder functions by @chiamp in https://github.com/google/flax/pull/2815
* Fixing the punctuation in the README by @arcAman07 in https://github.com/google/flax/pull/2811
* Implement lazy init by @jheek in https://github.com/google/flax/pull/2816
* improve nn.checkpoint docs by @cgarciae in https://github.com/google/flax/pull/2837
* Flax 0.6.5 point release by @levskaya in https://github.com/google/flax/pull/2856

**Full Changelog**: https://github.com/google/flax/compare/v0.6.4...v0.6.5

<hr /><em>This discussion was created from the release <a href='https://github.com/google/flax/releases/tag/v0.6.5'>Version 0.6.5</a>.</em>

**Comment by hkjeon13** [comment]
This version has the problem that the FlaxModel from the huggingface can't use the GPUs on the Colab Notebook. I tried many ways based on the error message, but any solution didn't work except installing the previous version <=0.6.4. 

Here is my Colab Notebook script with GPU (just for the explanation).
```
#Flax-trainer is my PyPI module for the training, but it has no dependencies except jax, jaxlib, transformers, datasets, evaluate, and flax.
!pip install git+https://github.com/hkjeon13/flax-trainer.git
```
```
from flax_trainer import FlaxTrainerForSequenceClassification
from transformers import FlaxAutoModelForSequenceClassification, TrainingArguments, AutoTokenizer
from datasets import load_dataset


tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
dataset = load_dataset("glue", "cola", split="train")
print(dataset)
def example_function(examples):
    tokenized_inputs = tokenizer(
        examples["sentence"], 
        padding='max_length', 
        max_length=256, 
        truncation=True
    )
    if "label" in examples:
        tokenized_inputs["labels"] = examples["label"]
    return tokenized_inputs

dataset = dataset.map(example_function, batched=True, remove_columns=dataset.column_names)

model = FlaxAutoModelForSequenceClassification.from_pretrained("bert-base-cased", from_pt=True)
args = TrainingArguments("runs")
trainer = FlaxTrainerForSequenceClassification(
    model=model,
    args = args,
    train_dataset=dataset
)

trainer.train()
```
I tested on the GPU env, but flax (==0.6.5) didn't recognize the GPU resource.
I also do the updates on jax, jaxlib for the flax(==0.6.5), but it occurs dependencies problem.

**Reply by levskaya** [reply]
Hi - could you give any more information on what the error messages were?

There's very little in flax itself that would affect GPU compatibility.

I'm guessing there might be an issue with recent jax / jaxlib that we require with new minimum version requirements?

We can try to track down what's happening, but more info on the errors would be helpful.

**Reply by cgarciae** [reply]
I managed to get `transformers` working on colab (cpu) by updating `jax` before installing flax + transformers:

```
! pip install -U jax jaxlib
! pip install flax transformers[flax]
```

**Reply by hkjeon13** [reply]
Hi, Thanks for your concern. It is the best way for the error sharing my Colab Notebook.
Here is the notebook link on github gist: https://gist.github.com/hkjeon13/20bbf42637a75a1d29f5ea9e5276b2a3
In this notebook, you could find the error "WARNING:jax._src.lib.xla_bridge:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)" even on the GPU environment.

**Reply by andsteing** [reply]
Can you try updating the PIP packages like this?

```
jax[cuda11_cudnn82]>=0.4.2
--find-links https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
```

(from https://github.com/google-research/vision_transformer/blob/main/vit_jax/requirements.txt )

**Reply by hkjeon13** [reply]
> Can you try updating the PIP packages like this?
> 
> ```
> jax[cuda11_cudnn82]>=0.4.2
> --find-links https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
> ```
> 
> (from https://github.com/google-research/vision_transformer/blob/main/vit_jax/requirements.txt )

I tried this and got a successful training result. Thank you for the suggestion:)

**Comment by wookayin** [comment]
Why was 0.6.5 version yanked? I cannot find a relevant thread anywhere. 

https://pypi.org/project/flax/0.6.5/
